pact_broker/docker_configuration.rb

```diff
@@ -128,6 +132,35 @@ def to_s
       end
     end
 
+    def space_delimited_integer_list_or_default property_name
+      if env_populated?(property_name)
+        SpaceDelimitedIntegerList.parse(env(property_name))
+      else
+        default(property_name)
+      end
+    end
+
+    class SpaceDelimitedIntegerList < Array
+      def initialize list
+        super(list)
+      end
+
+      def self.parse(string)
+        array = (string || '').split(' ').filter { |word|
+          (Integer(word) rescue nil) != nil
```

Comment:
I'd personally pull this into an `integer?(string)` method.

Reply:
I moved it to `self.integer?` method. However, for some reasons I could adjust the filter method to same style like `collect`, what I have tried are:
```
.filter(self::integer)
.filter(self:integer?)
.filter(integer?)
.filter(&:integer?)
```

So I accomplish it using as it was the only way I found that compiles the files & passes the tests:
```
.filter { |word| integer?(word) }
```

Let me know if you see another point for improvement

Reply:
Yeah, the `&:method` trick only works on the object being iterated over. What you've done is perfect.